public/app/features/alerting/unified/components/contact-points/Modals.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { Button, Modal, ModalProps } from '@grafana/ui';
+
+type ModalHook<T = undefined> = [JSX.Element, (item: T) => void, () => void];
+
+/**
+ * This hook controls the delete modal for contact points, showing loading and error states when appropriate
+ */
+export const useDeleteContactPointModal = (
+  handleDelete: (name: string) => Promise<void>,
+  isLoading: boolean
+): ModalHook<string> => {
+  const [showModal, setShowModal] = useState(false);
+  const [contactPoint, setContactPoint] = useState<string>();
+  const [error, setError] = useState<unknown | undefined>();
+
+  const handleDismiss = useCallback(() => {
+    if (isLoading) {
+      return;
+    }
+
+    setContactPoint(undefined);
+    setShowModal(false);
+    setError(undefined);
+  }, [isLoading]);
+
+  const handleShow = useCallback((name: string) => {
+    setContactPoint(name);
+    setShowModal(true);
+    setError(undefined);
+  }, []);
+
+  const handleSubmit = useCallback(() => {
+    if (contactPoint) {
+      handleDelete(contactPoint)
+        .then(() => setShowModal(false))
+        .catch(setError);
+    }
+  }, [handleDelete, contactPoint]);
+
+  const modalElement = useMemo(() => {
```

Comment:
Not meant for this PR but maybe we could have a generic delete modal component customizable by its props, so that it's consistent in all the places where we want to delete items with modals.

Reply:
Definitely agree, we should standardize our modal behaviour across the board in alerting – I think I prefer the one we have here (disabling the submit button, and waiting) over what we have in the notification policies page.